WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationsViewController/NotificationsViewModel.swift

```diff
@@ -73,4 +95,34 @@ final class NotificationsViewModel {
                 .first(where: notMatcher)
         }
     }
+
+    // MARK: - Handling Inline Actions
+
+    func sharePostActionTapped(with notification: Notification) -> ShareablePost? {
+        guard let url = notification.url else {
+            self.crashLogger.logMessage("Failed to share a notification post due to null url", level: .error)
+            return nil
+        }
+        let content: ShareablePost = (
+            url: url,
+            title: notification.title
+        )
+        self.trackInlineActionTapped(action: .sharePost)
+        return content
+    }
+}
+
+// MARK: - Analytics Tracking
+
+private extension NotificationsViewModel {
+
+    func trackInlineActionTapped(action: InlineAction) {
+        self.analyticsTracker.track(.notificationsInlineActionTapped, properties: ["inline_action": action.rawValue])
```

Comment:
Is it recommended to extract such keys as `"inline_action"` to constants?

Reply:
Since the string `inline_action` is specific to this tracking event and is not used elsewhere nor anticipated to be used again, defining it as a constant may not be necessary. This keeps our code simple and avoids an overload of single-use constants.